home_work_8/index.js

```diff
@@ -0,0 +1,188 @@
+const validateValue = (value) => {
+  if (value === undefined || value === null) {
+    return;
+  }
+  return value;
+};
+const EMPTY_VALUE = "This value is not specified";
+
+//1 Створити масив, довжину та елементи якого задає користувач (через prompt).
+const testArray = [];
+
+for (let i = 1; i < 2; i++) {
+  const lengthOfArray = +prompt(
+    "Enter the length of the array as a number (from 1 to 10)"
+  );
+  if (isNaN(lengthOfArray) || lengthOfArray > 10) {
+    i--;
+  } else {
+    for (let i = 1; i <= lengthOfArray; i++) {
+      const enteredValue = +prompt(`Enter a numeric value №${i}`);
+      if (!isNaN(enteredValue)) {
+        testArray.push(enteredValue);
+      } else {
+        i--;
+      }
+    }
+  }
+}
+
+console.log(testArray);
+const sortedTestArray = testArray.sort((a, b) => a - b);
+
+console.log(sortedTestArray);
+
+/*
+3.0 Дано масив - список покупок. Кожен елемент масиву - це обʼєкт вигляду:
+{productName: 'bread', productPrice: 14.5, productQuantity: 2}.
+Мінімальний довжина такого масиву - 6
+*/
+
+const productsList = [
+  { productName: "bread", productPrice: 14.5, productQuantity: 2 },
+  { productName: "salo", productPrice: 120.5, productQuantity: 4 },
+  { productName: "garlic", productPrice: 60.5, productQuantity: 13 },
+  { productName: "carrots", productPrice: 14.5, productQuantity: 22 },
+  { productName: "cabbage", productPrice: 12.5, productQuantity: 44 },
+  { productName: "potatoes", productPrice: 6, productQuantity: 56 },
+];
+
+// 3.1 Порахувати загальну вартість корзини та вивести суму у зрозумілому форматі
+const totalProductPrice = productsList.reduce(
+  (sum, price) => sum + price.productPrice * price.productQuantity,
+  0
+);
+console.log(`Total cost: ${totalProductPrice}`);
+
+// 3.2 Порахувати найменшу кількість продукту, який потрібно купити
+const minQuantity = productsList.sort(
+  (el1, el2) => el1.productQuantity - el2.productQuantity
```

Comment:
As an option sorting can be done by
`(a, b) => (a.productPrice > b.productPrice ? 1 : -1)`

Reply:
noticed